### DISCUSSION TITLE
Constructor injected closable resource in a ParameterResolver

### DISCUSSION THREAD

**Discussion by roded** [junit-team/junit-framework#4081] [Q&A] [answered]
Hello,
I have a `TypeBasedParameterResolver` which I use to inject an object borrowed from a pool (FTP connections) into a test class' constructor. 

This test class's test methods are running in parallel, so for every test method, JUnit is creating a new instance of the test class and injecting it with a borrowed object.

I have a problem though: It seems like the `ExtensionContext` at the constructor injection point is relevant for the entire test class and not the specific test method. So I can't seem to get a method related context onto which to attach a `CloseableResource` so that when the method's exection is done, will return the borrowed object to the pool. Is this the intended behaviour: values injected into the constructor of a test class should be applicable to the test class context?

I imagine that this is an issue only for constructor injection and that if I used `@BeforeEach`, then the `ExtensionContext` will be applicable to the currently executing test method. However, I do enjoy my fields being `final`.

Any input much appreciated.
Cheers.

**Comment by marcphilipp** [comment, marked as answer]
Your description is accurate. In https://github.com/junit-team/junit5/issues/4057, we've just implemented an [opt-in solution](https://junit.org/junit5/docs/snapshot/api/org.junit.jupiter.api/org/junit/jupiter/api/extension/TestInstantiationAwareExtension.html#getTestInstantiationExtensionContextScope(org.junit.jupiter.api.extension.ExtensionContext)) to address this issue, but it's not released, yet.
